src/DocumentFragment.php

```diff
@@ -21,4 +21,50 @@
 class DocumentFragment extends \DOMDocumentFragment {
 use LiveProperty, ParentNode;
 
+/**
+ * @param string $id
+ * @return Element|null
+ */
+public function getElementById(string $id)/*:?Element*/ {
+	return $this->callMethodOnTemporaryElement(
+		"querySelector", ["#" . $id]);
+}
+
+/**
+ * @param string $selectors CSS selector(s)
+ * @return Element|null
+ */
+public function querySelector(string $selectors) {
+    return $this->callMethodOnTemporaryElement(
+        "querySelector", [$selectors]);
+}
+
+/**
+ * @param string $selectors CSS selector(s)
+ * @return HTMLCollection
+ */
+public function querySelectorAll(string $selectors):HTMLCollection {
+    return $this->callMethodOnTemporaryElement(
+        "querySelectorAll", [$selectors]);
+}
+
+private function callMethodOnTemporaryElement(string $methodName, array $args) {
+	$childNodes = [];
+
+    $tempElement = $this->ownerDocument->createElement("document-fragment");
+
+	while($this->firstChild) {
```

Comment:
DOM Nodes can _only_ exist in one place within one DOM tree. On line 57, after the node has been stored in the array, it is appended to the temporary element which in fact removes it from its current location.  This may be a case for some comments, explaining the counter intuitive but standard behaviour.